prow/crier/controller.go

```diff
@@ -157,23 +163,26 @@ func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 
 func (r *reconciler) reconcile(ctx context.Context, log *logrus.Entry, req reconcile.Request) (*reconcile.Result, error) {
 
+	pjContext, pjCancel := context.WithTimeout(ctx, 30*time.Second)
+	defer pjCancel()
 	var pj prowv1.ProwJob
-	if err := r.pjclientset.Get(ctx, req.NamespacedName, &pj); err != nil {
+	if err := r.pjclientset.Get(pjContext, req.NamespacedName, &pj); err != nil {
```

Comment:
This just retrieves it from the cache, I don't think that logic can even respect a context

Reply:
The controller runtime client accepts the context so I assumed it was respected. I think it makes sense to supply something besides the background context so that we don't have to make assumptions about the implementation of the client, but I'm happy to omit the stricter 30 second timeout or switch back to the background context if you don't think this is reasonable.

Reply:
Yeah, I mostly would like to avoid the additional subcontext here to keep things simple, the global timeout is a good thing to have and there is no reason not to use that context here

Reply:
+1 to there being a single `reconcile()` level context with a guaranteed reasonable/conservative timeout -- 30m? 1hr?

This obviates the need for `updateContext, pjContext` and should also be used for `ShouldReport(), Report()` etc.

Reply:
👍 Switched to a single 30m context at the `reconcile()` level.